kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/aggregate/AggregateIdentifierConverter.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.kotlin.aggregate
+
+import java.util.*
+import java.util.function.Function
+
+/**
+ * Defines a converter from a string to custom identifier type.
+ * @param [ID] type of aggregate identifier.
+ *
+ * @author Simon Zambrovski
+ * @since 0.2.0
+ */
+interface AggregateIdentifierConverter<ID> : Function<String, ID> {
```

Comment:
Here are my thoughts on this...

1. Initially this feature is coming from Jan Hendrik, he has implemented it in Scala too.
2. In contrast to usual aggregate having a default parameterless constructor where the type of the aggregate identifier doesn't really matter, in case we pass the aggregate identifier as string (defined by AxonFramework's `AggregateFactory#createAggregateRoot`) we need to take care of the conversion of the String to the aggregate identifier parameter of the constructor.
3. I believe that building this feature without the converter will slightly easier, but I believe that the usage of String as aggregate identifier is a (unneeded) limitation and it will be hard to change this later on.
4. As you can see, I provided easy access methods for `String`, `UUID` and `<T>` types of aggregate identifier.